elastic/eui
# [EuiComboBox] Fix `onSearchChange` bug with incorrect `hasMatchingOptions` data

## Summary

closes https://github.com/elastic/eui/issues/6638

The bug was occurring because `onSearchChange` was being called before the internal `matchingOptions` state had a chance to update (via `getDerivedStateFromProps`). Moving the callback to after `this.setState({ searchValue })` is done updating fixes the order of operations.

> ⚠️  code diffs are mostly from yarn.lock (storybook upgrade), actual fix is super small - please [follow along by commit](https://github.com/elastic/eui/pull/7334/commits)!

## QA

![onSearchChange](https://github.com/elastic/eui/assets/549407/2c688c5e-3422-46d8-9974-84866c257dc6)

- Go to https://eui.elastic.co/pr_7334/storybook/?path=/story/euicombobox--playground
- Go to the `Actions` tab
- [x] Confirm that when typing to search, the second `true/false` arg matches what you see visually with the combobox options list (true if any items are listed, false if none)

### General checklist

- Browser QA - N/A
- Docs site QA - N/A
- Code quality checklist
- Release checklist
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist - N/A
